src/electron/main/main.ts

```diff
@@ -81,7 +81,7 @@ const createWindow = () => {
     mainWindow.on('closed', () => {
         // Drop all references to the window object, to force garbage collection
         ipcMessageDispatcher.unregisterMessageSink(mainWindowMessageSink);
-        mainWindow = null;
+        mainWindow = null!;
```

Comment:
Doesn't the ! postfix operator throw an assertion? That seems like an odd choice for something that is part of the typical shutdown...

Reply:
No; the `!` is called the "non-null assertion operator", but I think you're misunderstanding what the word "assertion" implies in the context of a "type assertion". The [docs for the operator](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-) are a good starting point; the usage here is essentially telling typescript "I know this variable's type suggests it can't be null, but please allow me to pass it a possibly-null (here: definitely null) value anyway". I'm doing this here because it's specifically in the context of the on-closed event, where we know nothing will be referencing it afterwards; the alternative would be to make the type of `mainWindow` nullable, but that felt worse because it forced unnecessary checks everywhere the variable is used.